app/scripts/lib/channels/notifier.js

```diff
@@ -20,12 +20,12 @@ define(function (require, exports, module) {
     CHANGE_PASSWORD: {
       name: 'fxaccounts:change_password',
       schema: {
-        email: 'String',
-        keyFetchToken: 'String',
-        sessionToken: 'String',
-        uid: 'String',
-        unwrapBKey: 'String',
-        verified: 'Boolean'
+        email: Vat.email().required(),
+        keyFetchToken: Vat.keyFetchToken().optional(),
```

Comment:
Ahha...  Yes, now I understand...

`keyFetchToken` and `unwrapBKey` are undefined. The [original validation logic](https://github.com/mozilla/fxa-content-server/pull/4981/files#diff-6ed00eca06c5d167fddeb62e3b9ebbf9L181) iterated over all the keys in `data`, not in the schema. If a key wasn't in the data, then it was never validated for validity.

Based on that, ISTM the original validation logic may have been subtly incorrect and the change from `required` to `optional` is needed. That jives with what I thought when making the update, glad to know why the change is necessary.